File: Python/dealership.py
import time
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class Dealership(object):
    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 30)

    # Make the browser use the current location
    def MyLocation(self):
        time.sleep(2)
        self.driver.save_screenshot('Dealership_Location.png')
        self.wait.until(EC.visibility_of_element_located((By.XPATH, '//button[text()="Select this Dealer"]')))
        self.driver.find_elements_by_xpath('//button[text()="Select this Dealer"]')[0].click()
        #self.driver.find_element_by_xpath('//button[@class="btn btn-primary dealerSelectBtn___8r46A"]').click()

File: Java/src/PageFactory/Dealership.java
package PageFactory;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.concurrent.TimeUnit;

public class Dealership
{
    WebDriver driver;
    WebDriverWait wait;

    @FindBy(xpath="//button[text()=\"Select this Dealer\"]")
    WebElement DealerLocation;

    public Dealership(WebDriver driver, WebDriverWait wait)
    {
        this.driver = driver;
        this.wait = wait;
        PageFactory.initElements(driver, this);
    }

    public void DealerLocation() throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//button[text()=\"Select this Dealer\"]")));
        DealerLocation.click();
    }
}
